### PR TITLE
Ability to close index instead of delete

### PR DESCRIPTION
**Pull request by oleg-andreyev**
In case if you don't want to delete old index (backup for emergency case) you should close it.
Adding `--no-delete` option to `PopulateCommand`

### CODE REVIEW

File: Index/AliasProcessor.php [hunk 1]
```diff
@@ -154,6 +162,30 @@ private function deleteIndex(Client $client, $indexName)
...
     /**
+     * Close an index
+     *
+     * @param Client $client
+     * @param string $indexName
+     */
+    private function closeIndex(Client $client, $indexName)
+    {
+        try {
+            $path = sprintf("%s/_close", $indexName);
```
Comment (oleg-andreyev) [inline Index/AliasProcessor.php]: We should use: `\FOS\ElasticaBundle\Elastica\Index`.

File: EventListener/PopulateListener.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    /**
+     * @param IndexPopulateEvent $event
+     */
+    public function onPostIndexPopulate(IndexPopulateEvent $event)
+    {
+        $this->resetter->switchIndexAlias($event->getIndex(), $event->getOption('delete'));
+    }
+}
```
Comment (solivier) [inline EventListener/PopulateListener.php]: missing new line at the end of the file

Comment (oleg-andreyev) [inline EventListener/PopulateListener.php]: Don't know why github is trimming last line in diff, but if you'll checkout branch you'll see it's there.

**Comment by oleg-andreyev** [comment]
@merk what do you think?

**Comment by XWB** [comment]
Thanks.

File: Command/PopulateCommand.php [hunk 3]
```diff
@@ -90,11 +91,16 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $index = $input->getOption('index');
         $type = $input->getOption('type');
         $reset = !$input->getOption('no-reset');
+        $delete = !$input->getOption('no-delete');
+
         $options = array(
+            'delete'        => $delete,
```
Comment (thanosp) [inline Command/PopulateCommand.php]: Adding these as `$options` seems to break during the strict validation of options later on https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Provider/AbstractProvider.php#L65.
`reset` is already there but `delete` is missing.

Comment (oleg-andreyev) [inline Command/PopulateCommand.php]: Ah. Seems that branch was quite old. I'll prepare fix for it ASAP

Comment (oleg-andreyev) [inline Command/PopulateCommand.php]: @thanosp `default` options are defined in `configureOptions`
see https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Provider/AbstractProvider.php#L118

**Pull request merged**